frontend/app/services/EventbriteService.scala

```diff
@@ -103,6 +103,14 @@ abstract class LiveService extends EventbriteService {
   val gridService = GridService(Config.gridConfig.url)
   val contentApiService = GuardianContentService
 
+  def mkRichEvent(event: EBEvent): Future[RichEvent] = for {
+    gridImageOpt <- gridImageFor(event)
+    props = (event, gridImageOpt, contentApiService.content(event.id))
+  } yield this match {
+    case GuardianLiveEventService => GuLiveEvent.tupled(props)
+    case LocalEventService => LocalEvent.tupled(props)
```

Comment:
Probably a bad idea, sadly... you're trying to work out what subclass you're in, better to just let them live in the subclasses.